sympy/functions/special/polynomials.py

```diff
@@ -1088,12 +1088,14 @@ class laguerre(OrthogonalPolynomial):
 
     @classmethod
     def eval(cls, n, x):
+        if n.is_integer is False:
+            raise ValueError("Error: n should be an integer.")
         if not n.is_Number:
             # Symbolic result L_n(x)
             # L_{n}(-x)  --->  exp(-x) * L_{-n-1}(x)
             # L_{-n}(x)  --->  exp(x) * L_{n-1}(-x)
-            if n.could_extract_minus_sign():
-                return exp(x) * laguerre(n - 1, -x)
+            if n.could_extract_minus_sign() and not(-n - S.One).could_extract_minus_sign():
```

Comment:
Since n is an Expr it is not necessary to write 1 as S.One here, above and below.
```suggestion
            if n.could_extract_minus_sign() and not(-n - 1).could_extract_minus_sign():
```